packages/bot-web-ui/src/stores/flyout-store.js

```diff
@@ -22,64 +22,65 @@ export default class FlyoutStore {
-     * @param {Element[]} xml_list list of XML nodes
-     * @memberof FlyoutStore
-     */
     @action.bound
-    setContents(xml_list, search_term = '') {
-        const text_limit = 20;
-        const processed_xml = xml_list;
-
-        this.block_listeners.forEach(listener => Blockly.unbindEvent_(listener));
-        this.block_workspaces.forEach(workspace => workspace.dispose());
-        this.block_listeners = [];
-        this.block_workspaces = [];
-
-        this.is_help_content = false;
-        this.search_term = search_term.length > text_limit ? `${search_term.substring(0, text_limit)}...` : search_term;
-        this.flyout_content = xml_list;
-
-        this.setFlyoutWidth(processed_xml);
-        this.setVisibility(true);
+    onMount() {
+        this.workspace = Blockly.derivWorkspace;
+        this.initFlyout();
+        window.addEventListener('click', this.onClickOutsideFlyout);
+    }
 
-        // apparently setFlyoutWidth doesn't calculate blocks dimentions until they're visible
-        // using setTimeout is a workaround to solve this issue
-        setTimeout(() => this.setFlyoutWidth(processed_xml), 50);
+    @action.bound
+    onUnmount() {
+        window.removeEventListener('click', this.onClickOutsideFlyout);
     }
 
-    /**
-     * Sets whether the flyout is visible or not.
-     * @param {boolean} is_visible
-     * @memberof FlyoutStore
-     */
     @action.bound
-    setVisibility(is_visible) {
-        this.is_visible = is_visible;
+    initFlyout() {
+        const workspace = this.workspace;
 
-        if (!is_visible) {
-            this.flyout_content = [];
+        const options = {
+            parentWorkspace: workspace,
+            RTL: workspace.RTL,
+            horizontalLayout: workspace.horizontalLayout,
+        };
+
+        if (workspace.horizontalLayout) {
+            this.flyout = new Blockly.HorizontalFlyout(options);
+        } else {
+            this.flyout = new Blockly.VerticalFlyout(options);
         }
+
+        this.flyout.targetWorkspace_ = workspace;
+        this.flyout.workspace_.targetWorkspace = workspace;
+
+        // A flyout connected to a workspace doesn't have its own current gesture.
+        this.flyout.workspace_.getGesture = this.flyout.targetWorkspace_.getGesture.bind(this.flyout.targetWorkspace_);
+
+        // Get variables from the main workspace rather than the target workspace.
+        this.workspace.variableMap_ = this.flyout.targetWorkspace_.getVariableMap();
+
+        this.flyout.workspace_.createPotentialVariableMap();
+
+        workspace.flyout_ = this.flyout;
     }
 
-    /**
-     * Sets whether the flyout is search or not.
-     * @param {boolean} is_search
-     * @memberof FlyoutStore
-     */
     @action.bound
-    setIsSearchFlyout(is_search) {
-        this.selected_category = '';
-        this.is_search_flyout = is_search;
+    initFlyoutButton(flyout_button) {
+        const callbackKey = flyout_button.getAttribute('callbackKey');
```

Comment:
How about using snake case?
```suggestion
        const callback_key = flyout_button.getAttribute('callbackKey');
```

Reply:
This is done 👍